daemon/k8s_watcher.go

```diff
@@ -1817,6 +1817,7 @@ func (d *Daemon) updateK8sNodeTunneling(k8sNodeOld, k8sNodeNew *types.Node) erro
 		Source: ipcache.FromKubernetes,
 	})
 	if !selfOwned {
+		d.nodeDiscovery.Manager.NodeSoftUpdated(*nodeNew)
```

Comment:
Newbie question: how come the datapath does not have to be updated here? E.g., why use `NodeSoftUpdated` vs. `NodeUpdated`?

Reply:
I believe that the `ipcache.IPIdentityCache.Upsert()` would already trigger such a datapath update. 

If I follow, the problem here is that we previously wouldn't issue any notification to the manager that the node has potentially changed, so then when a health IP shows up, this code wouldn't propagate that information through to the API (and then through to `cilium-health`, which would then ping the new IP instead of the old one)